### PR TITLE
Don't retry SemanticException

### PR DESCRIPTION
**Pull request by webrouse**
Jira: https://keboola.atlassian.net/browse/COM-681

Changes:
- Hive's `SemanticException` is not retried.

### CODE REVIEW

**Comment by webrouse** [comment]
@pivnicek Can you make review please?I would like to close ZD ticket.

**Comment by pivnicek** [comment]
Sorry Michal, I was on 3 other things :-D  Checking it now

File: src/Connection/HiveRetryPolicy.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+use Retry\RetryContextInterface;
+
+class HiveRetryPolicy extends SimpleRetryPolicy
+{
+    public function canRetry(RetryContextInterface $context): bool
+    {
+        $e = $context->getLastException();
+
+        // Don't retry "SemanticException"
+        if ($e && strpos($e->getMessage(), 'SemanticException') !== false) {
```
[the remark this answers is not in the document]
Comment (pivnicek) [reply src/Connection/HiveRetryPolicy.php:17]: This will do for now, until we eventually get the retry by error code thing going :-)

**Review by pivnicek — APPROVED** [review]
LGTM 👍

**Comment by webrouse** [comment]
Thank you! 
Yes, I know that there is a lot of work on ZD. That's why I want the next service to have one ticket less ;)

**Pull request merged**